addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -161,14 +339,114 @@ class ComposerTextInput extends Component {
+                    ev.stopPropagation();
+                } else {
+                    this._onKeyupTextareaEscape(ev);
+                }
                 break;
-            default:
+            // ENTER, HOME, END, UP, DOWN, PAGE UP, PAGE DOWN, TAB: check if navigation in mention propositions
+            case 'Enter':
+                if (this.state.isMentionsOpen) {
+                    this._insertMentionSuggestion(active);
+                }
                 break;
+            case 'ArrowUp':
+            case 'PageUp':
+                if (this.state.isMentionsOpen) {
+                    if (selectionIndex !== 0) {
+                        active = this.state.mentionSuggestions[0][selectionIndex - 1];
+                    } else {
+                        active = this.state.mentionSuggestions[0][this.state.mentionSuggestions[0].length - 1];
+                    }
+                    this.state.active_id = active.id;
+                }
+                break;
+            case 'ArrowDown':
+            case 'PageDown':
+                if (this.state.isMentionsOpen) {
+                    if (selectionIndex !== this.state.mentionSuggestions[0].length - 1) {
+                        active = this.state.mentionSuggestions[0][selectionIndex + 1];
+                    } else {
+                        active = this.state.mentionSuggestions[0][0];
+                    }
+                    this.state.active_id = active.id;
+                }
+                break;
+            case 'Home':
+                if (this.state.isMentionsOpen) {
+                    active = this.state.mentionSuggestions[0][0];
+                    this.state.active_id = active.id;
+                }
+                break;
+            case 'End':
+                if (this.state.isMentionsOpen) {
+                    active = this.state.mentionSuggestions[0][this.state.mentionSuggestions[0].length - 1];
+                    this.state.active_id = active.id;
+                }
+                break;
+            case 'Tab':
+                if (this.state.isMentionsOpen) {
+                    if (ev.shiftKey) {
+                        if (selectionIndex === 0) {
+                            active = this.state.mentionSuggestions[0][this.state.mentionSuggestions[0].length - 1];
+                        }
+                        else {
+                            active = this.state.mentionSuggestions[0][selectionIndex - 1];
+                        }
+                    } else {
+                        if (selectionIndex === this.state.mentionSuggestions[0].length - 1) {
+                            active = this.state.mentionSuggestions[0][0];
+                        }
+                        else {
+                            active = this.state.mentionSuggestions[0][selectionIndex + 1];
+                        }
+                    }
+                    this.state.active_id = active.id;
+                }
```

Comment:
long case handler could be isolated in functions, like here `_onKeyupTextareaTab()`